tinyhackathon/submission.py

```diff
@@ -124,6 +133,36 @@ def upload_submission(
         console.print(f"[red]Error: All completions must be unique. Found {duplicate_count} duplicate completions.[/red]")
         raise typer.Exit(code=1)
 
+    # Check if weight_class is provided, and if not, try to get it from existing metadata or prompt
+    metadata_remote_path = f"submissions/{username}/metadata.json"
+    existing_metadata = None
+
+    if weight_class is None:
+        # Try to get existing weight_class from metadata.json
+        try:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                api.hf_hub_download(repo_id=hf_repo, repo_type="dataset", filename=metadata_remote_path, local_dir=tmp_dir)
+                existing_metadata_path = Path(tmp_dir) / metadata_remote_path.split("/")[-1]
+                if existing_metadata_path.exists():
+                    with open(existing_metadata_path, "r") as f:
+                        existing_metadata = json.load(f)
+                        if "weight_class" in existing_metadata and existing_metadata["weight_class"]:
+                            existing_weight_class = existing_metadata["weight_class"]
+                            console.print(f"[yellow]Using existing weight class: {existing_weight_class}[/yellow]")
+                            weight_class = WeightClass(existing_weight_class)
+        except:
+            # If we can't get the metadata, we'll prompt the user
+            pass
+
+        # If still no weight_class from metadata, prompt the user
+        if weight_class is None:
+            weight_class = typer.prompt(
+                "Select model weight class size (small: up to 30M, medium: up to 60M, or large: up to 120M)",
+                type=WeightClass,
+                default=WeightClass.MEDIUM,
```

Comment:
In the command line this actually shows up as `Select model weight class size (small: up to 30M, medium: up to 60M, or large: up to 120M) [WeightClass.MEDIUM]` suggesting to the user that `WeightClass.MEDIUM` is a valid option. I think this should just be changed to `WeightClass.MEDIUM.value`, so it does not show up this way.

Reply:
This is one of the few typer functions which doesn't automatically handle string enums nicely. Fixed.